Paintroid/src/main/java/org/catrobat/paintroid/iotasks/LoadImage.kt

```diff
@@ -0,0 +1,110 @@
+/*
+ * Paintroid: An image manipulation application for Android.
+ * Copyright (C) 2010-2021 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.paintroid.iotasks
+
+import android.content.ContentResolver
+import android.content.Context
+import android.net.Uri
+import android.util.Log
+import android.webkit.MimeTypeMap
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import org.catrobat.paintroid.FileIO
+import java.io.IOException
+import java.lang.ref.WeakReference
+import java.util.Locale
+
+class LoadImage(
+    callback: LoadImageCallback,
+    private val requestCode: Int,
+    private val uri: Uri?,
+    context: Context,
+    private val scaleImage: Boolean,
+    private val scopeIO: CoroutineScope
+) {
+    private val callbackRef: WeakReference<LoadImageCallback> = WeakReference(callback)
+    private val context: WeakReference<Context> = WeakReference(context)
+
+    fun execute() {
```

Comment:
Too many nested code lines. 
Please refactor.

Reply:
Could please take a quick look at it again, I've tried to make new functions and reduced the nested lines :)